Pull Request For #29 Warnings

LeightonJonker opened the pull request:
Closes Issue #29 .
Created different modals for different situations. Senses if files have been modified and alerts users of impending progress loss if they wish to either exit or refresh the application, or sign into another account. (modal appears on pressing of the sign in button.)
In addition, small changes need to be made if issue fix is to be linked with #7 (sign out button leading back to the home page). These changes would namely be what function the "Sign out" button leads to.

Additionally, if a user has committed changes but not pushed them, a modal will also notify them of possible lost progress if they wish to exit / refresh the application or log out / log into another account.

@softeng-701 For Pull Requesting

Thanks!

LeightonJonker committed:
Warnings Draft Implementation

The product of adding warnings to VisualGit in the case of users wanting to Exit / Sign Out / Pull if there are commits that have not been commited and if the repository has not been pushed.

LeightonJonker committed:
Added option to reload

As opposed to only exiting the application when prompted; added the feature of allowing users to choose between reloading or exiting application, along with a back button.

LeightonJonker committed:
#29 Warning feature implementation

softeng-701 commented:
@LeightonJonker Please squash your commits. Also, another reviewer is needed

chaomix reviewed (approved):
Just a few nitpicks: 
In modal1W Committed & Pushed should be lower case.
In modalW2 same thing & 'if' should be capitalised.
In modalW3 same thing & "Pull" should be lower case.
Remove some old/unnecessary comments.
Otherwise, looks good to me (until issue #7 is fixed?)

LeightonJonker committed:
Added grammar changes + Commenting

As reviewed by Chaomix; changed capitals for modal text and removed unnecessary comments while added additional comments; next will be to squash commits and merge.

LeightonJonker committed:
Finishing Touches Warnings

LeightonJonker committed:
Merge branch 'master' into master

softeng-701 commented:
@LeightonJonker As you probably realized its against the workflow of this assignment to merge your own PR. Can you look at the merge conflicts in the revert, thanks
